// playwright-bdd/dist/gen/specialTags.js
"use strict";
/**
 * Special tags.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpecialTags = void 0;
class SpecialTags {
    constructor(ownTags) {
        this.ownTags = ownTags;
        this.extractFlags();
        this.extractRetries();
        this.extractTimeout();
        this.extractMode();
    }
    extractFlags() {
        if (this.ownTags.includes(`@slow`))
            this.slow = true;
        if (this.ownTags.includes(`@fail`))
            this.fail = true;
        // order is important
        const executionFlags = ['only', 'skip', 'fixme'];
        for (const flag of executionFlags) {
            if (this.ownTags.includes(`@${flag}`)) {
                this[flag] = true;
                return;
            }
        }
    }
    extractRetries() {
        for (const tag of this.ownTags.reverse()) {
            const match = tag.match(/@retries:(\d+)/i);
            if (match) {
                this.retries = Number(match[1]);
                return;
            }
        }
    }
    extractTimeout() {
        for (const tag of this.ownTags.reverse()) {
            const match = tag.match(/@timeout:(\d+)/i);
            if (match) {
                this.timeout = Number(match[1]);
                return;
            }
        }
    }
    extractMode() {
        for (const tag of this.ownTags.reverse()) {
            const match = tag.match(/@mode:(default|parallel|serial)/i);
            if (match) {
                this.mode = match[1];
                return;
            }
        }
    }
}
exports.SpecialTags = SpecialTags;
